src/scheduler.h

```diff
@@ -9,29 +9,33 @@
 
 static constexpr int32_t SCHEDULER_MINTICKS = 50;
 
+class SchedulerTask;
+
+using SchedulerTask_ptr = std::unique_ptr<SchedulerTask>;
+
 class SchedulerTask : public Task
 {
 public:
+	SchedulerTask(uint32_t delay, TaskFunc&& f) : Task(std::forward<TaskFunc>(f)), delay(delay) {}
```

Comment:
must become public to give abllity to call c-tor by make_unique from ```SchedulerTask_ptr createSchedulerTask(uint32_t delay, TaskFunc&& f)```

Reply:
you could use a friend class

Reply:
Friend is here but somehow it's not compile :<

Reply:
the problem is make_unique becouse it has no access to private c-tor of SchedulerTask class
there is no way to use something like as friend:
```template <typename... Args> Friend std::unique_ptr<SchedulerTask> std::make_unique(Args&&... args);```

Reply:
`friend` sucks, just make it public